### PR TITLE
CI: Bump clang-format version to 17

### PR DESCRIPTION
**Pull request by davidebeatrici**

### CODE REVIEW

**Comment by davidebeatrici** [comment]
Interesting... there are a few inconsistencies between `clang-format` 18.1.8 on my machine and Ubuntu's 18.1.3.

I wonder what the best course of action would be at this point...

**Comment by Krzmbrzl** [comment]
The first question is: what is your plan to update all currently open PRs to the new formatting to avoid tons of merge conflicts?

If you don't have a solution for that (I don't - in theory it is clear what to do but in practice?I have not yet found a reasonable way to actually do it) I'm against updating.

**Comment by Krzmbrzl** [comment]
> I wonder what the best course of action would be at this point...

Don't use v18 - use v17 instead which should not receive further minor releases (I think?)

**Comment by davidebeatrici** [comment]
> The first question is: what is your plan to update all currently open PRs to the new formatting to avoid tons of merge conflicts?
> 
> If you don't have a solution for that (I don't - in theory it is clear what to do but in practice?I have not yet found a reasonable way to actually do it) I'm against updating.

I will take care of rebasing any pull requests that are affected and otherwise ready to be merged.

Should be feasible considering we currently have 36 open PRs, including this one. And with a bit of luck many of them are probably not going to require any changes.

> Don't use v18 - use v17 instead which should not receive further minor releases (I think?)

Right. From https://releases.llvm.org:

File: overlay/HardHook_minhook.cpp [hunk 1]
```diff
@@ -65,7 +65,7 @@ void HardHook::setup(voidFunc func, voidFunc replacement) {
 void HardHook::setupInterface(IUnknown *unkn, LONG funcoffset, voidFunc replacement) {
 	fods("HardHook: setupInterface: Replacing %p function #%ld", unkn, funcoffset);
 	void **ptr = reinterpret_cast< void ** >(unkn);
-	ptr        = reinterpret_cast< void ** >(ptr[0]);
+	ptr        = reinterpret_cast< void        **>(ptr[0]);
```
Comment (davidebeatrici) [inline overlay/HardHook_minhook.cpp:68]: Weird alignment.

File: overlay/HardHook_x86.cpp [hunk 2]
```diff
@@ -116,7 +116,7 @@ void *HardHook::cloneCode(void **porig) {
 	// end of the chain. Hence these jumps need not be part of the trampoline.
 	while (*o == 0xe9) { // JMP
 		unsigned char *tmp = o;
-		int *iptr          = reinterpret_cast< int * >(o + 1);
+		int *iptr          = reinterpret_cast< int          *>(o + 1);
```
Comment (davidebeatrici) [inline overlay/HardHook_x86.cpp:119]: Weird alignment.

File: overlay/HardHook_x86.cpp [hunk 3]
```diff
@@ -205,7 +205,7 @@ void *HardHook::cloneCode(void **porig) {
 
 	// Add a relative jmp back to the original code, to after the copied code
 	n[idx++]              = 0xe9;
-	int *iptr             = reinterpret_cast< int * >(&n[idx]);
+	int *iptr             = reinterpret_cast< int             *>(&n[idx]);
```
Comment (davidebeatrici) [inline overlay/HardHook_x86.cpp:208]: Weird alignment.

File: overlay/HardHook_x86.cpp [hunk 4]
```diff
@@ -276,7 +276,7 @@ void HardHook::setup(voidFunc func, voidFunc replacement) {
 void HardHook::setupInterface(IUnknown *unkn, LONG funcoffset, voidFunc replacement) {
 	fods("HardHook: setupInterface: Replacing %p function #%ld", unkn, funcoffset);
 	void **ptr = reinterpret_cast< void ** >(unkn);
-	ptr        = reinterpret_cast< void ** >(ptr[0]);
+	ptr        = reinterpret_cast< void        **>(ptr[0]);
```
Comment (davidebeatrici) [inline overlay/HardHook_x86.cpp:279]: Weird alignment.

File: overlay/olsettings.cpp [hunk 5]
```diff
@@ -120,7 +120,7 @@ static int getModeInternal() {
 	}
 
 	DWORD sz = sizeof(mode);
-	err      = RegQueryValueExA(key, "mode", nullptr, nullptr, reinterpret_cast< BYTE * >(&mode), &sz);
+	err      = RegQueryValueExA(key, "mode", nullptr, nullptr, reinterpret_cast< BYTE      *>(&mode), &sz);
```
Comment (davidebeatrici) [inline overlay/olsettings.cpp:123]: Weird alignment.

File: src/Channel.cpp [hunk 6]
```diff
@@ -28,7 +28,7 @@ Channel::Channel(unsigned int id, const QString &name, QObject *p) : QObject(p)
 	bInheritACL = true;
 	uiMaxUsers  = 0;
 	bTemporary  = false;
-	cParent     = qobject_cast< Channel * >(p);
+	cParent     = qobject_cast< Channel     *>(p);
```
Comment (davidebeatrici) [inline src/Channel.cpp:31]: Weird alignment.

File: src/HostAddress.cpp [hunk 7]
```diff
@@ -126,8 +126,8 @@ std::string HostAddress::toStdString() const {
 }
 
 bool HostAddress::isV6() const {
-	std::uint64_t firstEightBytes      = *(reinterpret_cast< const std::uint64_t * >(m_byteRepresentation.data()));
-	std::uint16_t bytesNineAndTen      = *(reinterpret_cast< const std::uint16_t * >(&m_byteRepresentation[8]));
+	std::uint64_t firstEightBytes      = *(reinterpret_cast< const std::uint64_t      *>(m_byteRepresentation.data()));
+	std::uint16_t bytesNineAndTen      = *(reinterpret_cast< const std::uint16_t      *>(&m_byteRepresentation[8]));
```
Comment (davidebeatrici) [inline src/HostAddress.cpp:130]: Weird alignment.

File: src/MumbleProtocol.cpp [hunk 8]
```diff
@@ -636,7 +639,7 @@ namespace Protocol {
 
 					// Virtual array entries 1 and 2 are actually a single uint64. Note that the timestamp is
 					// whatever the client sent to the server and thus it does not require an endian-transformation
-					m_pingData.timestamp           = *reinterpret_cast< const std::uint64_t * >(&dataArray[1]);
+					m_pingData.timestamp           = *reinterpret_cast< const std::uint64_t           *>(&dataArray[1]);
```
Comment (davidebeatrici) [inline src/MumbleProtocol.cpp:642]: Weird alignment.

File: src/mumble/AudioOutputSpeech.cpp [hunk 9]
```diff
@@ -209,7 +209,7 @@ void AudioOutputSpeech::addFrameToBuffer(const Mumble::Protocol::AudioData &audi
 	// Since a storageIndex of zero will look the same as a null pointer, we always add one to
 	// ensure the library never erroneously confuses this entry with a free slot.
 	JitterBufferPacket jbp;
-	jbp.data      = reinterpret_cast< char * >(storageIndex) + 1;
+	jbp.data      = reinterpret_cast< char      *>(storageIndex) + 1;
```
Comment (davidebeatrici) [inline src/mumble/AudioOutputSpeech.cpp:212]: Weird alignment.

File: src/mumble/Cert.cpp [hunk 10]
```diff
@@ -517,11 +517,11 @@ QByteArray CertWizard::exportCert(const Settings::KeyPair &kp) {
 	QByteArray key = kp.second.toDer();
 	QByteArray qba;
 
-	p    = reinterpret_cast< const unsigned char * >(key.constData());
+	p    = reinterpret_cast< const unsigned char    *>(key.constData());
```
Comment (davidebeatrici) [inline src/mumble/Cert.cpp:520]: Weird alignment.

File: src/mumble/Cert.cpp [hunk 11]
```diff
@@ -517,11 +517,11 @@ QByteArray CertWizard::exportCert(const Settings::KeyPair &kp) {
...
 	pkey = d2i_AutoPrivateKey(nullptr, &p, key.length());
 
 	if (pkey) {
-		p    = reinterpret_cast< const unsigned char * >(crt.constData());
+		p    = reinterpret_cast< const unsigned char    *>(crt.constData());
```
Comment (davidebeatrici) [inline src/mumble/Cert.cpp:524]: Weird alignment.

File: src/mumble/Cert.cpp [hunk 12]
```diff
@@ -535,7 +535,7 @@ QByteArray CertWizard::exportCert(const Settings::KeyPair &kp) {
 			foreach (const QSslCertificate &cert, qlCerts) {
 				X509 *c = nullptr;
 				crt     = cert.toDer();
-				p       = reinterpret_cast< const unsigned char * >(crt.constData());
+				p       = reinterpret_cast< const unsigned char       *>(crt.constData());
```
Comment (davidebeatrici) [inline src/mumble/Cert.cpp:538]: Weird alignment.

File: src/mumble/JackAudio.cpp [hunk 13]
```diff
@@ -600,7 +600,7 @@ void JackAudioSystem::ringbufferWriteAdvance(jack_ringbuffer_t *buffer, const si
 }
 
 int JackAudioSystem::processCallback(jack_nframes_t frames, void *) {
-	auto const input  = dynamic_cast< JackAudioInput * >(Global::get().ai.get());
+	auto const input  = dynamic_cast< JackAudioInput  *>(Global::get().ai.get());
```
Comment (davidebeatrici) [inline src/mumble/JackAudio.cpp:603]: Weird alignment.

File: src/mumble/JackAudio.cpp [hunk 14]
```diff
@@ -618,7 +618,7 @@ int JackAudioSystem::processCallback(jack_nframes_t frames, void *) {
 }
 
 int JackAudioSystem::sampleRateCallback(jack_nframes_t, void *) {
-	auto const input  = dynamic_cast< JackAudioInput * >(Global::get().ai.get());
+	auto const input  = dynamic_cast< JackAudioInput  *>(Global::get().ai.get());
```
Comment (davidebeatrici) [inline src/mumble/JackAudio.cpp:621]: Weird alignment.

File: src/mumble/JackAudio.cpp [hunk 15]
```diff
@@ -633,7 +633,7 @@ int JackAudioSystem::sampleRateCallback(jack_nframes_t, void *) {
 }
 
 int JackAudioSystem::bufferSizeCallback(jack_nframes_t frames, void *) {
-	auto const input  = dynamic_cast< JackAudioInput * >(Global::get().ai.get());
+	auto const input  = dynamic_cast< JackAudioInput  *>(Global::get().ai.get());
```
Comment (davidebeatrici) [inline src/mumble/JackAudio.cpp:636]: Weird alignment.

File: src/mumble/OverlayClient.cpp [hunk 16]
```diff
@@ -151,10 +151,10 @@ void OverlayClient::updateMouse() {
 			QImage outmask(img.width(), h, QImage::Format_MonoLSB);
 
 			for (int i = 0; i < h; ++i) {
-				const quint32 *srcimg  = reinterpret_cast< const quint32 * >(img.scanLine(i + h));
+				const quint32 *srcimg  = reinterpret_cast< const quint32  *>(img.scanLine(i + h));
```
Comment (davidebeatrici) [inline src/mumble/OverlayClient.cpp:154]: Weird alignment.

File: src/mumble/OverlayClient.cpp [hunk 17]
```diff
@@ -151,10 +151,10 @@ void OverlayClient::updateMouse() {
...
 				const quint32 *srcmask = reinterpret_cast< const quint32 * >(img.scanLine(i));
 
-				quint32 *dstimg  = reinterpret_cast< quint32 * >(out.scanLine(i));
+				quint32 *dstimg  = reinterpret_cast< quint32  *>(out.scanLine(i));
```
Comment (davidebeatrici) [inline src/mumble/OverlayClient.cpp:157]: Weird alignment.

File: src/mumble/PulseAudio.cpp [hunk 18]
```diff
@@ -197,7 +197,7 @@ void PulseAudioSystem::eventCallback(pa_mainloop_api *api, pa_defer_event *) {
 	AudioOutputPtr ao     = Global::get().ao;
 	AudioInput *raw_ai    = ai.get();
 	AudioOutput *raw_ao   = ao.get();
-	PulseAudioInput *pai  = dynamic_cast< PulseAudioInput * >(raw_ai);
+	PulseAudioInput *pai  = dynamic_cast< PulseAudioInput  *>(raw_ai);
```
Comment (davidebeatrici) [inline src/mumble/PulseAudio.cpp:200]: Weird alignment.

File: src/mumble/ServerHandler.cpp [hunk 19]
```diff
@@ -354,7 +356,7 @@ void ServerHandler::setServerSynchronized(bool synchronized) {
 }
 
 void ServerHandler::hostnameResolved() {
-	ServerResolver *sr                    = qobject_cast< ServerResolver * >(QObject::sender());
+	ServerResolver *sr                    = qobject_cast< ServerResolver                    *>(QObject::sender());
```
Comment (davidebeatrici) [inline src/mumble/ServerHandler.cpp:359]: Weird alignment.

File: src/mumble/UserView.cpp [hunk 20]
```diff
@@ -157,7 +157,7 @@ void UserView::mouseReleaseEvent(QMouseEvent *evt) {
 
 	QModelIndex idx = indexAt(clickPosition);
 	if ((evt->button() == Qt::LeftButton) && idx.isValid()) {
-		UserModel *userModel   = qobject_cast< UserModel * >(model());
+		UserModel *userModel   = qobject_cast< UserModel   *>(model());
```
Comment (davidebeatrici) [inline src/mumble/UserView.cpp:160]: Weird alignment.

File: src/mumble/WASAPI.cpp [hunk 21]
```diff
@@ -472,7 +472,7 @@ void WASAPIInput::run() {
 			wfe.Format.nAvgBytesPerSec = wfe.Format.nBlockAlign * wfe.Format.nSamplesPerSec;
 
 			micpwfxe = &wfe;
-			micpwfx  = reinterpret_cast< WAVEFORMATEX * >(&wfe);
+			micpwfx  = reinterpret_cast< WAVEFORMATEX  *>(&wfe);
```
Comment (davidebeatrici) [inline src/mumble/WASAPI.cpp:475]: Weird alignment.

File: src/murmur/Server.cpp [hunk 22]
```diff
@@ -689,7 +689,7 @@ void Server::udpActivated(int socket) {
 	uint8_t controldata[CMSG_SPACE(std::max(sizeof(struct in6_pktinfo), sizeof(struct in_pktinfo)))];
 
 	memset(&msg, 0, sizeof(msg));
-	msg.msg_name       = reinterpret_cast< struct sockaddr * >(&from);
+	msg.msg_name       = reinterpret_cast< struct sockaddr       *>(&from);
```
Comment (davidebeatrici) [inline src/murmur/Server.cpp:692]: Weird alignment.

File: src/murmur/Server.cpp [hunk 23]
```diff
@@ -707,9 +707,9 @@ void Server::udpActivated(int socket) {
 #else
 	int fromlen = static_cast< int >(sizeof(from));
 	SOCKET sock = static_cast< SOCKET >(socket);
-	len         = ::recvfrom(sock, reinterpret_cast< char * >(m_udpDecoder.getBuffer().data()),
-                     static_cast< int >(m_udpDecoder.getBuffer().size()), 0,
-                     reinterpret_cast< struct sockaddr * >(&from), &fromlen);
+	len         = ::recvfrom(sock, reinterpret_cast< char         *>(m_udpDecoder.getBuffer().data()),
+							 static_cast< int >(m_udpDecoder.getBuffer().size()), 0,
+							 reinterpret_cast< struct sockaddr         *>(&from), &fromlen);
```
Comment (davidebeatrici) [inline src/murmur/Server.cpp:712]: Weird alignment.

File: src/murmur/Server.cpp [hunk 24]
```diff
@@ -844,7 +844,7 @@ void Server::run() {
 				uint8_t controldata[CMSG_SPACE(std::max(sizeof(struct in6_pktinfo), sizeof(struct in_pktinfo)))];
 
 				memset(&msg, 0, sizeof(msg));
-				msg.msg_name       = reinterpret_cast< struct sockaddr * >(&from);
+				msg.msg_name       = reinterpret_cast< struct sockaddr       *>(&from);
```
Comment (davidebeatrici) [inline src/murmur/Server.cpp:847]: Weird alignment.

File: src/murmur/Server.cpp [hunk 25]
```diff
@@ -1079,7 +1079,7 @@ void Server::sendMessage(ServerUser &u, const unsigned char *data, int len, QByt
 		memset(controldata, 0, sizeof(controldata));
 
 		memset(&msg, 0, sizeof(msg));
-		msg.msg_name    = reinterpret_cast< struct sockaddr * >(&u.saiUdpAddress);
+		msg.msg_name    = reinterpret_cast< struct sockaddr    *>(&u.saiUdpAddress);
```
Comment (davidebeatrici) [inline src/murmur/Server.cpp:1082]: Weird alignment.

File: src/murmur/Server.cpp [hunk 26]
```diff
@@ -1660,9 +1660,9 @@ void Server::connectionClosed(QAbstractSocket::SocketError err, const QString &r
 		qhUsers.remove(u->uiSession);
 		qhHostUsers[u->haAddress].remove(u);
 
-		quint16 port = (u->saiUdpAddress.ss_family == AF_INET6)
-						   ? (reinterpret_cast< sockaddr_in6 * >(&u->saiUdpAddress)->sin6_port)
-						   : (reinterpret_cast< sockaddr_in * >(&u->saiUdpAddress)->sin_port);
+		quint16 port                             = (u->saiUdpAddress.ss_family == AF_INET6)
+													   ? (reinterpret_cast< sockaddr_in6                             *>(&u->saiUdpAddress)->sin6_port)
+													   : (reinterpret_cast< sockaddr_in                             *>(&u->saiUdpAddress)->sin_port);
```
Comment (davidebeatrici) [inline src/murmur/Server.cpp:1665]: Weird alignment.

File: src/tests/Benchmark.cpp [hunk 27]
```diff
@@ -209,7 +209,7 @@ void Client::readyRead() {
 			unsigned char b[6];
 			ssl->read(reinterpret_cast< char * >(b), 6);
 
-			ptype    = qFromBigEndian(*reinterpret_cast< quint16 * >(&b[0]));
+			ptype    = qFromBigEndian(*reinterpret_cast< quint16    *>(&b[0]));
```
Comment (davidebeatrici) [inline src/tests/Benchmark.cpp:212]: Weird alignment.

File: src/tests/TestCrypt/TestCrypt.cpp [hunk 28]
```diff
@@ -185,9 +185,9 @@ void TestCrypt::authcrypt() {
 		const unsigned char rawkey[AES_BLOCK_SIZE] = { 0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07,
 													   0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f };
 		const unsigned char nonce[AES_BLOCK_SIZE]  = { 0xff, 0xee, 0xdd, 0xcc, 0xbb, 0xaa, 0x99, 0x88,
-                                                      0x77, 0x66, 0x55, 0x44, 0x33, 0x22, 0x11, 0x00 };
+													   0x77, 0x66, 0x55, 0x44, 0x33, 0x22, 0x11, 0x00 };
 		std::string rawkey_str = std::string(reinterpret_cast< const char * >(rawkey), AES_BLOCK_SIZE);
-		std::string nonce_str  = std::string(reinterpret_cast< const char * >(nonce), AES_BLOCK_SIZE);
+		std::string nonce_str  = std::string(reinterpret_cast< const char  *>(nonce), AES_BLOCK_SIZE);
```
Comment (davidebeatrici) [inline src/tests/TestCrypt/TestCrypt.cpp:190]: Weird alignment.

File: src/tests/TestCrypt/TestCrypt.cpp [hunk 29]
```diff
@@ -219,9 +219,9 @@ void TestCrypt::xexstarAttack() {
 	const unsigned char rawkey[AES_BLOCK_SIZE] = { 0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07,
 												   0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f };
 	const unsigned char nonce[AES_BLOCK_SIZE]  = { 0xff, 0xee, 0xdd, 0xcc, 0xbb, 0xaa, 0x99, 0x88,
-                                                  0x77, 0x66, 0x55, 0x44, 0x33, 0x22, 0x11, 0x00 };
-	std::string rawkey_str                     = std::string(reinterpret_cast< const char * >(rawkey), AES_BLOCK_SIZE);
-	std::string nonce_str                      = std::string(reinterpret_cast< const char * >(nonce), AES_BLOCK_SIZE);
+												   0x77, 0x66, 0x55, 0x44, 0x33, 0x22, 0x11, 0x00 };
+	std::string rawkey_str                     = std::string(reinterpret_cast< const char                     *>(rawkey), AES_BLOCK_SIZE);
+	std::string nonce_str                      = std::string(reinterpret_cast< const char                      *>(nonce), AES_BLOCK_SIZE);
```
Comment (davidebeatrici) [inline src/tests/TestCrypt/TestCrypt.cpp:224]: Weird alignment.

File: src/tests/TestCrypt/TestCrypt.cpp [hunk 30]
```diff
@@ -278,9 +278,9 @@ void TestCrypt::tamper() {
...
```
Comment (davidebeatrici) [inline src/tests/TestCrypt/TestCrypt.cpp:283]: Weird alignment.

**Review by davidebeatrici — COMMENTED** [review]
llvm/llvm-project#105898

**Comment by Krzmbrzl** [comment]
Do we want to live with the weird pointer alignment or do we want to wait with the clang-format upgrade until v19 has been released (or rather v20 has been released, meaning that v19 doesn't receive updates anymore and is therefore stable)?

**Comment by davidebeatrici** [comment]
We have the weird pointer alignment in at least one other project of ours, but it was accidental.

I would wait until the fix is merged and then install the tool from LLVM's official repository: https://apt.llvm.org

**Comment by Krzmbrzl** [comment]
That's not compatible with the current CI pipeline and would likely also make the tool self-updating which we don't want.

**Comment by Krzmbrzl** [comment]
> Why not?

Because the action we are using uses docker containers. We can change the action of course but we might as well stick to it as it works and we don't have to maintain it.

> The idea is to always stick to the current version, as long as there are no PRs that are about to be merged.

We essentially always have open PRs so this doesn't make sense. We should just pick a version and stick to it and then update in a couple of years.

**Comment by Krzmbrzl** [comment]
Given that we'll have to wait for clang-format 19, I think this PR can be closed for now.

**Pull request closed without merging**